### PR TITLE
refactor: move SpecialStoragePolicy from brightray to atom

### PR DESCRIPTION
**Pull request by miniak**
##### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

notes: no-notes

### CODE REVIEW

**Comment by codebytere** [comment]
@miniak why was this done?

**Comment by miniak** [comment]
@codebytere I was experimenting with removing brightray as an unnecessary layer since it was merged into the main repository. This one was the easiest one to move. I guess it requires a broader discussion whether we want to do that.

**Comment by deepak1556** [comment]
> I was experimenting with removing brightray as an unnecessary layer since it was merged into the main repository. This one was the easiest one to move. I guess it requires a broader discussion whether we want to do that.

Yes we definitely want to do that, there has been discussion about it in the past https://github.com/electron/maintainers/issues/172 , some of the trivial components like this one and few others in https://github.com/electron/electron/pull/13956 can be moved as they are seen fit. As to other parts like app initialization, browser context and devtools bindings needs to be moved in a way thats non disruptive to other refactors like GN build, chromium upgrades, OOIF `<webview>`. Maybe have an umbrella issue to track those and discuss how we want to move them in, probably reuse components from chrome layer with patches rather than copies.

File: atom/browser/atom_browser_context.h [hunk 1]
```diff
@@ -77,6 +83,7 @@ class AtomBrowserContext : public brightray::BrowserContext {
   std::unique_ptr<AtomDownloadManagerDelegate> download_manager_delegate_;
   std::unique_ptr<WebViewManager> guest_manager_;
   std::unique_ptr<AtomPermissionManager> permission_manager_;
+  std::unique_ptr<SpecialStoragePolicy> storage_policy_;
```
Comment (deepak1556) [inline atom/browser/atom_browser_context.h]: Should be `scoped_refptr` as `storage::SpecialStoragePolicy` is ref counted.

Comment (miniak) [reply atom/browser/atom_browser_context.h]: @deepak1556 fixed

File: atom/browser/special_storage_policy.h [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
 #include "storage/browser/quota/special_storage_policy.h"
 
-namespace brightray {
+namespace atom {
 
 class SpecialStoragePolicy : public storage::SpecialStoragePolicy {
  public:
   SpecialStoragePolicy();
+  ~SpecialStoragePolicy() override;
```
Comment (deepak1556) [inline atom/browser/special_storage_policy.h]: can this line be reverted to fix chromium-style error in the gn build bots

```
../../electron/atom/browser/special_storage_policy.cc:11:1: error: [chromium-style] Classes that are ref-counted should have destructors that are declared protected or private.
SpecialStoragePolicy::~SpecialStoragePolicy() {}
^
../../electron/atom/browser/special_storage_policy.h:12:30: note: [chromium-style] 'SpecialStoragePolicy' inherits from 'storage::SpecialStoragePolicy' here
class SpecialStoragePolicy : public storage::SpecialStoragePolicy {
                             ^
../../storage/browser/quota/special_storage_policy.h:25:7: note: [chromium-style] 'SpecialStoragePolicy' inherits from 'base::RefCountedThreadSafe<SpecialStoragePolicy>' here
    : public base::RefCountedThreadSafe<SpecialStoragePolicy> {
```

**Comment by miniak** [comment]
@ckerr I've reverted the license change

**Comment by miniak** [comment]
@deepak1556 I've reverted the destructor change as requested

**Comment by codebytere** [comment]
@miniak can you add information to the PR description for posterity please? As a part of the new `release-notes` bot, you're also required to add `node: <description of fix>` in addition to the PR body description and below the checklist

**Pull request merged**